### ISSUE TITLE
Mathjax custom macros starting with non-alphabetical characters do not render correctly.

### ISSUE THREAD

**Issue by simurgh9** [remarkjs/react-markdown#698]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

react@18.0.0, react-dom@18.0.0, react-scripts@4.0.0, react-markdown@8.0.3

### Link to runnable example

I made a [CodeSandBox](https://codesandbox.io/s/affectionate-oskar-d5izyo?file=/src/index.js).

### Steps to reproduce

I am using React Markdown (`<ReactMarkdown children={markdown} />`) to render some markdown files that have some MathJax. Note that I am not using the remark or rehype plug-ins for the MathJax support. It is loaded using a script tag and typesetting is done each time `componentDidUpdate()` is called. Everything renders fine except the macros that start with a non alphabetical character. E. g.,

`$\newcommand{\?}{\stackrel{?}{=}} a\? b$.`

This renders: _a?b_

However, if I use an extra backslash after "a",

` $\newcommand{\?}{\stackrel{?}{=}} a\\? b$.`

Then the appropriate rendering is achieved. Similarly, if the macro is defined starting with an alphabetical character, e. g.,

`$\newcommand{\isit}{\stackrel{?}{=}} a\isit b$.`

We get the correct rendering.

I wrote this little React Component for you to test this out,

```jsx
import { createRoot } from "react-dom/client";
import { Component } from "react";
import ReactMarkdown from "react-markdown";

const rootElement = document.getElementById("root");
const root = createRoot(rootElement);

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { readyToTypeset: false };
  }
  componentDidMount() {
    window.MathJax = { startup: { ready: () => {
      window.MathJax.startup.defaultReady();
      this.setState({ readyToTypeset: true });}}};
    let script = document.createElement("script");
    script.defer = true;
    script.src = "https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js";
    document.head.appendChild(script);
  }
  componentDidUpdate() {
    if (this.state.readyToTypeset)
      window.MathJax.typesetPromise();
  }
  render() {
    return <ReactMarkdown children={
      "$$\\newcommand{\\?}{\\stackrel{?}{=}} x \\? y$$" +
      "$$\\newcommand{\\?}{\\stackrel{?}{=}} x \\\\? y$$" +
      "$$\\newcommand{\\isit}{\\stackrel{?}{=}} x \\isit y$$"
    } />;
  }
}

root.render(<App />);
```

Here is a screenshot of the output,

![rep](https://user-images.githubusercontent.com/43047754/177080669-9110ddd6-2dd5-4d75-b712-5f6fcfa39958.png)

### Expected behavior

You can goto https://math.stackexchange.com/ and click on "Ask Question" and copy:

```
$$\newcommand{\?}{\stackrel{?}{=}} x \? y$$
$$\newcommand{\isit}{\stackrel{?}{=}} x \isit y$$
```

This gets you the expected behaviour,

![rep1](https://user-images.githubusercontent.com/43047754/177081288-bd951923-97a7-43e5-ba76-d596da193436.png)

I have also tried putting a markdown file on a Github Pages website and accessing it that way, we still get the expected behaviour.

### Actual behavior

A markdown file containing `$$\newcommand{\?}{\stackrel{?}{=}} x \? y$$` renders _a?b_. See Steps to reproduce for details. 

### Runtime

Node v16

### Package manager

npm 8

### OS

Linux

### Build and bundle tools

Create React App

**Comment by wooorm** [comment]
You are running into how escaping things work.
First you are running into how escapes work in JavaScript: to place a double quote in a string marked with double quotes, you need to escape that double quote. By extension, you can’t just place a slash either, you also have to escape that.
Markdown also interprets slashes followed by punctuation as escapes.

The solution is: add more escapes.
Alternatively, use `remark-math`. It is better.

**Comment by simurgh9** [comment]
I need to be able to use a custom build of MathJax, i. e., control the source of the script tag. I am not sure how to do that with `remark-math`. Is there a way to tell `react-markdown` to leave the MathJax delimited text alone? E. g., in dollar signs `$...$`? This is what the Jekyll and Stack Mathematics seem to be doing.

**Comment by wooorm** [comment]
remark-math is exactly that.
Please see the readme for more information

**Comment by simurgh9** [comment]
I have already tried `remark-math`. I must be doing something wrong then since if I install `remark-math`, it seems to disable the custom macros defined by the custom build making me think that is is typesetting using it's own typesetter than the one I loaded. Could you point me to the documentation I should read? Or, edit the code sandbox link?

**Comment by wooorm** [comment]
You're talking about something the remark-math plugin does not do. You seem to be talking about something rehype-mathjax does. You could not use that. Or you could configure that: https://github.com/remarkjs/remark-math/tree/main/packages/rehype-mathjax#browser

**Comment by simurgh9** [comment]
Take a look at the [sandbox](https://codesandbox.io/s/nice-buck-i1nngy?file=/src/index.js), I added `remark-math` and it's stopped typesetting altogether. Comment out the plug-ins prop and typesetting starts to work again.

**Comment by wooorm** [comment]
Correct, which is expected. See the link I provided above for the next step, or do it yourself.

**Comment by simurgh9** [comment]
Okay, so I have figured out this much,

```jsx
class App extends Component {
  customMathjax() {
    return rehypeMathjax({
      tex: {
        macros: {
          RR: "{\\bf R}",
          bold: ["{\\bf #1}", 1]
        }
      }
    });
  }

  render() {
    return (
      <ReactMarkdown
        rehypePlugins={[this.customMathjax]}
        remarkPlugins={[remarkMath]}
        children={String.raw`$\RR$`}
      />
    );
  }
}
```

This renders the custom macro `\RR` in the configs correctly. But I can not figure out how do I use these configs to instead load my custom script say, `https:forexample.com/my_custom_mathjax.js`.

Here is a [sandbox](https://codesandbox.io/s/nice-buck-i1nngy?file=/src/index.js) with the above component.

In Layman's terms, how do I make all the macros that work with the script tag `<script id="https:forexample.com/my_custom_mathjax.js"></script>`. Note that what is hosted at `https:forexample.com/my_custom_mathjax.js` is a combined component just like the `https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js`.

**Comment by wooorm** [comment]
There is no `customMathjax` in your sandbox. Why not use the browser version I mentioned several times?

Your question now resolves around how to combine MathJax with React.
This poses a giant problem: React does not want you to use random other scripts on the same content.
Such questions are better asked in MathJax and/or React communities.

I’m locking this. This issue tracker is for bugs in `react-markdown`.
